src/DGtal/geometry/tools/QuickHull.h

```diff
@@ -0,0 +1,1599 @@
+      for ( Index f = 0; f < facets.size(); ++f )
+        if ( deleted_facets.count( f ) == 0 ) {
+          bool ok = checkFacet( f );
+          nbok   += ok ? 1 : 0;
+          nb     += 1;
+        }
+      return nb == nbok;
+    }
+
+    /// @return 'true' iff the current set of facets encloses all the points
+    ///
+    /// @note Be careful, this function is much slower than computing
+    /// the convex hull. It can take a long time since its complexity
+    /// is \f$ O(n f ) \f$, where \a n is the number of input points
+    /// and \a f the number of facets.
+    bool checkHull()
+    {
+      Index nbok = 0;
+      // for ( Index v = 0; v < points.size(); ++v ) {
+      for ( auto v : processed_points ) {
+        bool ok = true;
+        for ( Index f = 0; f < facets.size(); ++f )
+          if ( deleted_facets.count( f ) == 0 ) {
+            if ( above( facets[ f ], points[ v ] ) ) {
+              ok = false;
+              trace.error() << "- bad vertex " << v << " " << points[ v ]
+                            << " dist="
+                            << ( kernel.height( facets[ f ].H, points[ v ] ) )
+                            << std::endl;
+              break;
+            }
+          }
+        nbok += ok ? 1 : 0;
+      }
+      if ( debug_level >= 2 ) {
+        trace.info() << nbok << "/"
+                  << processed_points.size() << " vertices inside convex hull."
+                  << std::endl;
+      }
+      if ( nbok != processed_points.size() ) myStatus = Status::InvalidConvexHull;
+      return nbok == processed_points.size();
+    }
+  
+    /// @}
+    // ------------------------ public datas --------------------------
+  public:
+    /// @name public datas
+    /// @{
+  
+  public:
+    /// Kernel that is duplicated for computing facet geometries.
+    mutable Kernel kernel;
+    /// debug_level from 0:no to 2
+    int debug_level; 
+    /// the set of points, indexed as in the array.
+    std::vector< Point > points;
+    /// the surjective mapping between the input range and the output
+    /// range used for convex hull computation.
+    IndexRange input2comp;
+    /// the injective mapping between the convex hull point range and
+    /// the input range.
+    IndexRange comp2input;
+    /// Points already processed (and within the convex hull).
+    IndexRange processed_points;
+    /// assignment of points to facets
+    std::vector< Index > assignment;
+    /// the current set of facets.
+    std::vector< Facet > facets;
+    /// set of deleted facets
+    std::set< Index > deleted_facets;
+    /// point index -> vertex index (or UNASSIGNED)
+    IndexRange p2v;
+    /// vertex index -> point index
+    IndexRange v2p;
+    /// Number of finite facets
+    Size nb_finite_facets;
+    /// Number of infinite facets (!= 0 only for specific kernels)
+    Size nb_infinite_facets;
+    /// Timings of the different phases: 0: init, 1: facets, 2: vertices.
+    std::vector< double > timings;
```

Comment:
std::array<double, 3> ?

Reply:
Sometimes you do not process till the vertices, but stop at facets, so only two timings are computed.